src/modules/admin/components/forms/FormPreview.tsx

```diff
@@ -0,0 +1,127 @@
+import AccountBoxIcon from '@mui/icons-material/AccountBox';
+import ListIcon from '@mui/icons-material/List';
+import { Grid } from '@mui/material';
+import React, { useMemo, useRef, useState } from 'react';
+import CommonToggle, { ToggleItem } from '@/components/elements/CommonToggle';
+import Loading from '@/components/elements/Loading';
+import {
+  CONTEXT_HEADER_HEIGHT,
+  STICKY_BAR_HEIGHT,
+} from '@/components/layout/layoutConstants';
+import PageTitle from '@/components/layout/PageTitle';
+import useSafeParams from '@/hooks/useSafeParams';
+import { useScrollToHash } from '@/hooks/useScrollToHash';
+import DynamicForm, {
+  DynamicFormRef,
+} from '@/modules/form/components/DynamicForm';
+import FormNavigation from '@/modules/form/components/FormNavigation';
+import DynamicView from '@/modules/form/components/viewable/DynamicView';
+import {
+  AlwaysPresentLocalConstants,
+  getInitialValues,
+  getItemMap,
+  getLeftNavItems,
+} from '@/modules/form/util/formUtil';
+import { useGetFormDefinitionFieldsForEditorQuery } from '@/types/gqlTypes';
+
+const FormPreview = () => {
+  const { formId } = useSafeParams() as { formId: string };
+  const {
+    data: { formDefinition } = {},
+    loading,
+    error,
+  } = useGetFormDefinitionFieldsForEditorQuery({
+    variables: { id: formId },
+  });
+
+  const formRef = useRef<DynamicFormRef>(null);
+
+  // This form preview is not client- or enrollment-specific, so just use the always-present local constants
+  const localConstants = AlwaysPresentLocalConstants;
+
+  useScrollToHash(loading, STICKY_BAR_HEIGHT + CONTEXT_HEADER_HEIGHT);
+
+  const initialValues = useMemo(() => {
+    return formDefinition?.definition
+      ? {
+          ...getInitialValues(formDefinition?.definition, localConstants),
+        }
+      : {};
+  }, [formDefinition?.definition, localConstants]);
+  const [formValues, setFormValues] = useState<object>(initialValues);
+
+  type PreviewMode = 'preview' | 'readOnly';
+  const toggleItems: ToggleItem<PreviewMode>[] = [
+    {
+      value: 'preview',
+      label: 'Preview Form',
+      Icon: ListIcon,
+    },
+    {
+      value: 'readOnly',
+      label: 'Read Only View',
+      Icon: AccountBoxIcon,
+    },
+  ];
```

Comment:
I don't think this needs to be inside the component (ie being re-defined every render)? I would move it up to line 26